careers/backend-assignment.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Backend assignment
+description: A simple task to evaluate your skillset
+---
+
+# Welcome to the Bornfight "Army assignment" instructions!
+
+You may have gotten this link during our selection process for becoming a part of our team. If so, read on, this is
+what you need to do!
+
+The assignment is as follows:
+
+- Create a mini battle program consisting of two armies.
+
+- Your solution should basically be a text-based war between these two armies.
+
+- Each army should consist of a number of soldiers that are passed as parameters.
+
+- The army sizes should be passed through a GET parameter, for example`?army1=50&army2=48`
+
+You are required to implement at least one idea of your own, that introduces an element of randomness (it could be
+generals, earthquakes, soldiers poisoning each other, priests, dragons... we'll leave that up to you).
+
+### <u>The solution must, in some way, return which army has won the battle.</u>
+
+Battle logic, army structure, data types, class structure, file structure, outputs (if there are any), the choice
+of framework or no framework at all, any new idea or feature - is completely up to you.
+
+Basically, we are trying to figure out how you’ll implement these features, since you are not tied to anything and
+can do your very best.
+
+### The solution you provide should reflect this specification and, although you may introduce your own new features and ideas, it is important that the initial brief is met.
+
+---
+
+## What will be evaluated in the solution:
+
+- Object-oriented design and architecture
+- Correct usage and implementation of design patterns
+- Structure and placement of files and classes
+- Code cleanliness
+- Code consistency
+- General elements of well written code
+
+#### If you choose to use a framework to solve this task, we will expect you to use the framework as it was designed, keeping in mind its rules, structure, best practices and standards.
+
+## What will NOT be evaluated:
+
+- Visual design of the solution
+- Visual layout of the solution
+- Frontend of the solution
+- Graphical interface of the solution (if you choose to make one)
+
+## How to submit your solution:
+
+You can submit your result in a zip file and send it through an email or use a GitHub/GitLab/BitBucket repo.
+
+## Extra points:
+
+- Write tests you deem important
+- Use types
+- A linter passing codebase
```

Comment:
@tonymrakovcic 
Maybe we could replace linter with phpstan since we are using it regularly ?

Reply:
keep in mind this is a backend assignment, used mostly for PHP but sometimes for backend JS also :)
So this is just "linter" to work on both of them

Reply:
Okay, I get it, thanks